### ISSUE TITLE
Existing XML schema cannot be found.

### ISSUE THREAD

**Issue by bobpiece** [DMTF/Redfish-Service-Validator#31]
I am running your tool against Redfish API exposed by reference Intel's implementation of RSD.
Script is reporting following problem:

```
sudo python3 RedfishServiceValidator.py
System Info: Abc
RedfishServiceValidator Config details: ('https://localhost:8443', 'user:myUser', './SchemaFiles/metadata', 'no CheckCert', 'Attempt for Online Schema')
Start time: 05/25/17 - 09:30:48

*** ServiceRoot, /redfish/v1
/usr/local/lib/python3.5/dist-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
#ServiceRoot.v1_1_1.ServiceRoot
/usr/local/lib/python3.5/dist-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
SchemaURI missing reference link: /redfish/v1/$metadata ServiceRoot
SchemaURI unsuccessful: /redfish/v1/$metadata
validateURI: No schema XML for #ServiceRoot.v1_1_1.ServiceRoot ServiceRoot
Elapsed time: 0:00:00
1
Counter({'passGet': 1, 'failSchema': 1})
Validation has failed: 1 problems found
```

I am using following configuration:
```
[Information]
Updated = May 8, 2017
Description = Redfish Service-Schema Conformance Tool 0.91

[SystemInformation]
TargetIP = localhost:8443
UserName = myUser
Password = myPassword
SystemInfo = Abc
[Options]
MetadataFilePath = ./SchemaFiles/metadata
UseSSL = True
LocalOnlyMode = False
CertificateCheck = Off
Session_UserName = Test1
Session_Password = TestPwd1

```

I have verified metadata exposed by my service and it seems that mentioned type is correctly defined there. Service root of my service looks like:

```
{

    "@odata.context": "/redfish/v1/$metadata#ServiceRoot",
    "@odata.id": "/redfish/v1",
    "@odata.type": "#ServiceRoot.v1_1_1.ServiceRoot",
    "Id": "ServiceRoot",
    "Name": "Service root",
    "Description": "POD Manager Service root",
    "RedfishVersion": "1.0.2",
    "UUID": "24b00548-eb4e-40c6-84c9-0064525837ac"
   ...
}
```

Metadata exposed under /redfish/v1/$metadata#ServiceRoot 
```
<?xml version="1.0" encoding="UTF-8"?><edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
  <edmx:Reference Uri="/redfish/v1/metadata/Org.OData.Core.V1.xml">
    <edmx:Include Alias="OData" Namespace="Org.OData.Core.V1"/>
  </edmx:Reference>

  <edmx:Reference Uri="/redfish/v1/metadata/ServiceRoot.xml">
    <edmx:Include Namespace="ServiceRoot.v1_1_1"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/IntelRackScaleOem.xml">
    <edmx:Include Namespace="Intel.Oem"/>
  </edmx:Reference>

  <edmx:DataServices>
    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="Service">
      <EntityContainer Extends="ServiceRoot.v1_1_1.ServiceContainer" Name="Service">
        <Annotation String="Default Entity Container" Term="OData.Description"/>
      </EntityContainer>
    </Schema>
  </edmx:DataServices>
</edmx:Edmx>

```

Metadata exposed under /redfish/v1/metadata/ServiceRoot.xml:
```
<?xml version="1.0" encoding="UTF-8"?><!-- Copyright 2014-2015 Distributed Management Task Force, Inc. (DMTF). All rights reserved.--><edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">

  <edmx:Reference Uri="/redfish/v1/metadata/Org.OData.Core.V1.xml">
    <edmx:Include Alias="OData" Namespace="Org.OData.Core.V1"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/Resource.xml">
    <edmx:Include Namespace="Resource"/>
    <edmx:Include Namespace="Resource.v1_0_0"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/RedfishExtensions.xml">
    <edmx:Include Alias="Redfish" Namespace="RedfishExtensions.v1_0_0"/>
    <edmx:Include Alias="Validation" Namespace="Validation.v1_0_0"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/EventService.xml">
    <edmx:Include Namespace="EventService"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/AccountService.xml">
    <edmx:Include Namespace="AccountService"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/ComputerSystemCollection.xml">
    <edmx:Include Namespace="ComputerSystemCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/ComposedNodeCollection.xml">
    <edmx:Include Namespace="ComposedNodeCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/ChassisCollection.xml">
    <edmx:Include Namespace="ChassisCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/ManagerCollection.xml">
    <edmx:Include Namespace="ManagerCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/TaskService.xml">
    <edmx:Include Namespace="TaskService"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/SessionService.xml">
    <edmx:Include Namespace="SessionService"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/Message.xml">
    <edmx:Include Namespace="Message"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/JsonSchemaFileCollection.xml">
    <edmx:Include Namespace="JsonSchemaFileCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/MessageRegistryFileCollection.xml">
    <edmx:Include Namespace="MessageRegistryFileCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/SessionCollection.xml">
    <edmx:Include Namespace="SessionCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/StorageServiceCollection.xml">
    <edmx:Include Namespace="StorageServiceCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/EthernetSwitchCollection.xml">
    <edmx:Include Namespace="EthernetSwitchCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/FabricCollection.xml">
    <edmx:Include Namespace="FabricCollection"/>
  </edmx:Reference>
  <edmx:Reference Uri="/redfish/v1/metadata/TelemetryService.xml">
    <edmx:Include Namespace="TelemetryService"/>
  </edmx:Reference>


  <edmx:DataServices>

    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot">

      <EntityType Abstract="true" BaseType="Resource.v1_0_0.Resource" Name="ServiceRoot">
        <Annotation String="This resource represents the root of the Redfish service, located at the \redfish\v1 URI.  As a hypermedia API, all other resources accessible through the Redfish interface on this device are linked directly or indirectly from the Service Root." Term="OData.Description"/>
        <Annotation String="This object represents the root Redfish service.  All values for resources described by this schema shall comply to the requirements as described in the Redfish specification." Term="OData.LongDescription"/>
      </EntityType>
    </Schema>

    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot.v1_0_0">

      <EntityContainer Name="ServiceContainer">
        <Singleton Name="Service" Type="ServiceRoot.ServiceRoot"/>
        <Singleton Name="Systems" Type="ComputerSystemCollection.ComputerSystemCollection"/>
        <Singleton Name="Nodes" Type="ComposedNodeCollection.ComposedNodeCollection"/>
        <Singleton Name="Chassis" Type="ChassisCollection.ChassisCollection"/>
        <Singleton Name="Managers" Type="ManagerCollection.ManagerCollection"/>
        <Singleton Name="Tasks" Type="TaskService.TaskService"/>
        <Singleton Name="AccountService" Type="AccountService.AccountService"/>
        <Singleton Name="SessionService" Type="SessionService.SessionService"/>
        <Singleton Name="EventService" Type="EventService.EventService"/>
        <Singleton Name="Registries" Type="MessageRegistryFileCollection.MessageRegistryFileCollection"/>
        <Singleton Name="JsonSchemas" Type="JsonSchemaFileCollection.JsonSchemaFileCollection"/>
        <Singleton Name="Sessions" Type="SessionCollection.SessionCollection"/>
        <Singleton Name="Services" Type="StorageServiceCollection.StorageServiceCollection"/>
      </EntityContainer>

      <EntityType BaseType="ServiceRoot.ServiceRoot" Name="ServiceRoot">
        <Annotation String="This object represents the root Redfish service." Term="OData.Description"/>
        <Annotation String="This object represents the root Redfish service.  All values for resources described by this schema shall comply to the requirements as described in the Redfish specification." Term="OData.LongDescription"/>
        <Property Name="RedfishVersion" Nullable="false" Type="Edm.String">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="The version of the Redfish service" Term="OData.Description"/>
          <Annotation String="The value of this string shall represent the version of the Redfish service.  The format of this string shall be of the format majorversion.minorversion.errata in compliance with Protocol Version section of the Redfish specification. " Term="OData.LongDescription"/>
          <Annotation String="^\d+\.\d+\.\d+$" Term="Validation.Pattern"/>
        </Property>
        <Property Name="UUID" Type="Resource.UUID">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="Unique identifier for a service instance. When SSDP is used, this value should be an exact match of the UUID value returned in a 200OK from an SSDP M-SEARCH request during discovery. " Term="OData.Description"/>
          <Annotation String="The value of this string shall represent the id of the Redfish service instance.  The format of this string shall be a 32-byte value in the form 8-4-4-4-12.  If SSDP is used, this value shall be an exact match of the UUID value returned in a 200OK from an SSDP M-SEARCH request during discovery. RFC4122 describes methods that can be used to create a UUID value. The value should be considered to be opaque. Client software should only treat the overall value as a universally unique identifier and should not interpret any sub-fields within the UUID." Term="OData.LongDescription"/>
        </Property>
        <NavigationProperty Name="Systems" Nullable="false" Type="ComputerSystemCollection.ComputerSystemCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Systems." Term="OData.Description"/>
          <Annotation String="This object shall only contain a reference to a collection of resources that comply to the Systems schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Nodes" Type="ComposedNodeCollection.ComposedNodeCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Nodes." Term="OData.Description"/>
          <Annotation String="This object shall only contain a reference to a collection of resources that comply to the Nodes schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Chassis" Nullable="false" Type="ChassisCollection.ChassisCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Chassis." Term="OData.Description"/>
          <Annotation String="This object shall only contain a reference to a collection of resources that comply to the Chassis schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Managers" Nullable="false" Type="ManagerCollection.ManagerCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Managers." Term="OData.Description"/>
          <Annotation String="This object shall only contain a reference to a collection of resources that comply to the Managers schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Tasks" Nullable="false" Type="TaskService.TaskService">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to the Task Service." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the TaskService schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="SessionService" Nullable="false" Type="SessionService.SessionService">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to the Sessions Service." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the SessionService schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="AccountService" Nullable="false" Type="AccountService.AccountService">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to the Account Service." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the AccountService schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="EventService" Nullable="false" Type="EventService.EventService">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to the EventService." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the EventService schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Registries" Nullable="false" Type="MessageRegistryFileCollection.MessageRegistryFileCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Registries." Term="OData.Description"/>
          <Annotation String="This object shall contain a reference to Message Registry." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="JsonSchemas" Nullable="false" Type="JsonSchemaFileCollection.JsonSchemaFileCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Json-Schema files." Term="OData.Description"/>
          <Annotation String="This object shall only contain a reference to a collection of resources that comply to the SchemaFile schema where the files are Json-Schema files." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="EthernetSwitches" Type="EthernetSwitchCollection.EthernetSwitchCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection Ethernet Switches." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the Ethernet Switch schema." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <NavigationProperty Name="Services" Type="StorageServiceCollection.StorageServiceCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="This is a link to a collection of Storage Services." Term="OData.Description"/>
          <Annotation String="The classes structure shall only contain a reference to a resource that complies to the StorageService schema" Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
        <Property Name="Links" Nullable="false" Type="ServiceRoot.v1_0_0.Links">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="Contains references to other resources that are related to this resource." Term="OData.Description"/>
          <Annotation String="The Links property, as described by the Redfish Specification, shall contain references to resources that are related to, but not contained by (subordinate to), this resource." Term="OData.LongDescription"/>
          <Annotation Term="Redfish.Required"/>
        </Property>
      </EntityType>

      <ComplexType BaseType="Resource.Links" Name="Links">
        <NavigationProperty Name="Sessions" Nullable="false" Type="SessionCollection.SessionCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="Link to a collection of Sessions" Term="OData.Description"/>
          <Annotation String="This property shall contain the link to a collection of Sessions." Term="OData.LongDescription"/>
          <Annotation Term="OData.AutoExpandReferences"/>
        </NavigationProperty>
      </ComplexType>

    </Schema>
    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot.v1_0_2">
      <EntityType BaseType="ServiceRoot.v1_0_0.ServiceRoot" Name="ServiceRoot"/>
    </Schema>
    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot.v1_0_3">
      <EntityType BaseType="ServiceRoot.v1_0_2.ServiceRoot" Name="ServiceRoot"/>
    </Schema>
    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot.v1_1_0">
      <EntityType BaseType="ServiceRoot.v1_0_2.ServiceRoot" Name="ServiceRoot">
        <NavigationProperty ContainsTarget="true" Name="Fabrics" Nullable="false" Type="FabricCollection.FabricCollection">
          <Annotation EnumMember="OData.Permission/Read" Term="OData.Permissions"/>
          <Annotation String="A link to a collection of all fabric entities." Term="OData.Description"/>
          <Annotation String="The referenced collection shall contain references to all Fabric instances." Term="OData.LongDescription"/>
        </NavigationProperty>
      </EntityType>
    </Schema>

    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="ServiceRoot.v1_1_1">
      <EntityContainer Extends="ServiceRoot.v1_0_0.ServiceContainer" Name="ServiceContainer">
        <Singleton Name="Fabrics" Type="FabricCollection.FabricCollection"/>
        <Singleton Name="TelemetryService" Type="TelemetryService.TelemetryService"/>
      </EntityContainer>

      <EntityType BaseType="ServiceRoot.v1_1_0.ServiceRoot" Name="ServiceRoot"/>
    </Schema>

  </edmx:DataServices>
</edmx:Edmx>

```
It seems that script doesn't work correctly for XML tags containing colon, for example: ```<edmx:reference>```, following call:  ``` refs = soup.find_all('edmx:reference') ``` doesn't find requested tags.

**Comment by tomasg2012** [comment]
The issue here is that specifically, this $metadata reference refers to "ServiceRoot.v1_1_1", but the URI frag on "/redfish/v1/$metadata#ServiceRoot" refers to the namespace "ServiceRoot", which is not in this $metadata.

A correction could be:
```
<edmx:Reference Uri="/redfish/v1/metadata/ServiceRoot.xml">
    <edmx:Include Namespace="ServiceRoot.v1_1_1"/>
    <edmx:Include Namespace="ServiceRoot"/>
</edmx:Reference>
```

Unrelated to the issue, the frag to the right of $metadata is defined in the spec (and the tool only supports such) to be 'MetadataUrl#ResourceType', ResourceType being "the fully qualified name of the unversioned resource type", such that it should be "ServiceRoot.ServiceRoot"  As it stands, the validator does not support other $metadata frags, however this example works with the tool.  The design of the validator to be stricter on these regards if the spec demands it, but it doesn't seem to notice this, thank you.

**Comment by mraineri** [comment]
In this case, the `@odata.context` property is following the more formal OData format where the context spells out the entity hierarchy for how the client reached the resource in relation to the entry point.

While this is valid, I should point out that we have moved away from this model in Redfish and strongly recommend others to use the more simplistic format for context.
